frontend/src/components/MapView/Layers/PointDataLayer/index.tsx

```diff
@@ -110,7 +112,7 @@ function PointDataLayer({ layer, before }: LayersProps) {
     updateHistory,
   ]);
 
-  if (!features || map?.getSource(layerId) || !queryDate) {
+  if (!features || !queryDate) {
```

Comment:
@HarryMytilinaios this change seems to fix the disappearing layers. But this might also break something else unexpectedly. Let's investigate a bit and see if we need another safeguard.